### PR TITLE
Eoir case status

### PR DESCRIPTION
**Pull request by droza0**
## Why is this PR needed?

Accompaniment program staff wanted to have access to EOIR case status for friends for assessment of risk to court hearings or check ins.

## Solution

This solution implements an `EOIR_CASE_STATUSES` constant on the Friend model similarly to the other status constants in the model with options that can be used by select elements. The constant stores statuses as an array with structure `[titlecased_status, status_in_snakecase]`. 

```ruby 
#app/models/friend.rb
class Friend < ApplicationRecord
#...
     EOIR_CASE_STATUSES = %w[case_pending
                              immigration_judge_ordered_removal
                              prior_voluntary_departure 
                              appeal pending 
                              motion_to_reopen_submitted].map { |status| [status.titlecase, status] }
#...
end
```

The basic friend edit form is available to admins at the url root_url_path/communities/nyc/admin/friends/1/edit (for the friend with id=1).  Select fields here default to blank unless that property of a friend is not null, in which case they default to that property's value. A select field was implemented with the same structure as the other select fields listing the possible EOIR case statuses. This select field is available to regional and non-regional primary and non primary admins. The field is placed directly below the email field as it is high priority.

```erb
#app/views/admin/friends/_basic_form_fields.html.erb
<div class='row form-group'>
  <%= f.label :eoir_case_status, class: 'col-md-2 control-label' %>
  <div class='col-md-6'>
    <%= f.select :eoir_case_status, options_for_select(Friend::EOIR_CASE_STATUSES.map { |status_array| status_array[0] }, @friend.eoir_case_status), {include_blank: true}, {class: 'form-control'} %>
  </div>
</div>
```

The friends controller's friend_params method was updated in the same order as the select field is placed in the view:

```ruby
def FriendsController < ApplicationController
#...
 def friend_params
    params.require(:friend).permit(
      :first_name,
      :last_name,
      :middle_name,
      :email,
      :eoir_case_status,
      :phone,
      :a_number,
      :no_a_number,
      :ethnicity,
      :other_ethnicity,
      :gender,
      :date_of_birth,
      :status,
      :date_of_entry,
      :border_crossing_status,
      :border_queue_number,
      :notes,
      :asylum_status,
      :asylum_notes,
      :date_asylum_application_submitted,
      :lawyer_notes,
      :work_authorization_notes,
      :date_eligible_to_apply_for_work_authorization,
      :date_work_authorization_submitted,
      :work_authorization_status,
      :sijs_status,
      :date_sijs_submitted,
      :sijs_notes,
      :sijs_lawyer,
      :country_id,
      :lawyer_represented_by,
      :lawyer_referred_to,
      :zip_code,
      :jail_id,
      :criminal_conviction,
      :criminal_conviction_notes,
      :final_order_of_removal,
      :has_a_lawyer_for_detention,
      :bonded_out_by_nsc,
      :bond_amount,
      :date_bonded_out,
      :bonded_out_by,
      :city,
      :state,
      :date_foia_request_submitted,
      :foia_request_notes,
      :sponsor_name,
      :sponsor_phone_number,
      :sponsor_relationship,
      :intake_notes,
      :must_be_seen_by,
      :intake_date,
      :clinic_wait_list_priority,
      language_ids: [],
      user_ids: []
    ).merge(community_id: current_community.id, region_id: current_region.id)
  end
#...
end
```

The friend show page available to volunteers (for example at root_url_path/communities/nyc/friends/3) also displays the status in the same order as the basic edit form available to admins. This works for both primary and non-primary volunteers.

```erb
#app/views/friends/_info.html.erb
<h3>Basic</h3>

<p><strong>First Name:  </strong><%= friend.first_name %></p>
<p><strong>Middle Name:  </strong><%= friend.middle_name %></p>
<p><strong>Last Name:  </strong><%= friend.last_name %></p>
<p><strong>A Number:  </strong><%= friend.a_number %></p>
<p><strong>Phone:  </strong><%= friend.phone %></p>
<p><strong>Email:  </strong><%= friend.email %></p>
<p><strong>EOIR Case Status: </strong><%= friend.eoir_case_status %>
#...
```

### Link to associated issue: 

https://github.com/CZagrobelny/new_sanctuary_asylum/issues/243

### CODE REVIEW

File: app/views/admin/friends/_basic_form_fields.html.erb [hunk 1]
```diff
@@ -46,6 +46,13 @@
   </div>
 </div>
 
+<div class='row form-group'>
+  <%= f.label :eoir_case_status, class: 'col-md-2 control-label' %>
```
Comment (CZagrobelny) [inline app/views/admin/friends/_basic_form_fields.html.erb]: Could we add the same text for this label that you have in app/views/friends/_info.html.erb ?
Here's an example where we specify the text of a form label: https://github.com/CZagrobelny/new_sanctuary_asylum/blob/ca0f8be97f15bcd34a0598744c6cc02b9311daf6/app/views/admin/friends/_basic_form_fields.html.erb#L144

File: app/views/admin/friends/_basic_form_fields.html.erb [hunk 2]
```diff
@@ -46,6 +46,13 @@
   </div>
 </div>
 
+<div class='row form-group'>
```
Comment (CZagrobelny) [inline app/views/admin/friends/_basic_form_fields.html.erb]: Could we move :eoir_case_status just below the :status field on this page?

File: app/views/admin/friends/_basic_form_fields.html.erb (hunk 1 above)
Comment (droza0) [reply app/views/admin/friends/_basic_form_fields.html.erb]: Comitted!

File: app/views/admin/friends/_basic_form_fields.html.erb (hunk 2 above)
Comment (droza0) [reply app/views/admin/friends/_basic_form_fields.html.erb]: commited!

**Pull request merged**